scene/gui/code_edit.cpp

```diff
@@ -603,26 +603,15 @@ void CodeEdit::do_unindent() {
 	int cl = cursor_get_line();
 	const String &line = get_line(cl);
 
-	if (line[cc - 1] == '\t') {
-		_remove_text(cl, cc - 1, cl, cc);
+	if (line[0] == '\t') {
+		_remove_text(cl, 0, cl, 1);
 		cursor_set_column(MAX(0, cc - 1));
 		return;
-	}
-
-	if (line[cc - 1] != ' ') {
-		return;
-	}
-
-	int spaces_to_remove = _calculate_spaces_till_next_left_indent(cc);
-	if (spaces_to_remove > 0) {
-		for (int i = 1; i <= spaces_to_remove; i++) {
-			if (line[cc - i] != ' ') {
-				spaces_to_remove = i - 1;
-				break;
-			}
-		}
-		_remove_text(cl, cc - spaces_to_remove, cl, cc);
+	} else if (line[0] == ' ') {
+		int spaces_to_remove = _calculate_spaces_till_next_left_indent(get_first_non_whitespace_column(cl));
+		_remove_text(cl, 0, cl, spaces_to_remove);
 		cursor_set_column(MAX(0, cc - spaces_to_remove));
+		return;
```

Comment:
The two `return`s can be removed as they're both the last statements of a branch.